homeassistant/components/media_player/broadlink.py

```diff
@@ -0,0 +1,234 @@
+    vol.Optional(CONF_SOURCES, default={}): dict,
+    vol.Optional(CONF_DIGITS): DIGITS_SCHEMA,
+})
+
+SUPPORT_MAPPING = [
+    (CONF_COMMAND_ON, SUPPORT_TURN_ON),
+    (CONF_COMMAND_OFF, SUPPORT_TURN_OFF),
+    (CONF_VOLUME_UP, SUPPORT_VOLUME_STEP),
+    (CONF_VOLUME_DOWN, SUPPORT_VOLUME_STEP),
+    (CONF_VOLUME_MUTE, SUPPORT_VOLUME_MUTE),
+    (CONF_NEXT_TRACK, SUPPORT_NEXT_TRACK),
+    (CONF_PREVIOUS_TRACK, SUPPORT_PREVIOUS_TRACK),
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_devices,
+                               discovery_info=None):
+    """Set up platform."""
+    import broadlink
+
+    host = (config.get(CONF_HOST),
+            config.get(CONF_PORT))
+    mac = get_broadlink_mac(config.get(CONF_MAC))
+
+    link = broadlink.rm(
+        host,
+        mac,
+        None)
+
+    try:
+        await hass.async_add_job(link.auth)
+    except socket.timeout:
+        _LOGGER.warning("Timeout trying to authenticate to broadlink")
+        raise PlatformNotReady
+
+    async_add_devices([BroadlinkRM(link, config)])
+
+
+def get_supported_by_config(config):
+    """Calculate support flags based on available configuration entries."""
+    support = 0
+
+    for mapping in SUPPORT_MAPPING:
+        if mapping[0] in config:
+            support = support | mapping[1]
+
+    if config.get(CONF_SOURCES):
+        support = support | SUPPORT_SELECT_SOURCE
+
+    if config.get(CONF_DIGITS):
+        support = support | SUPPORT_PLAY_MEDIA
+
+    return support
+
+
+def get_broadlink_mac(mac: str):
+    """Convert a mac address string with : in it to just a flat string."""
+    return binascii.unhexlify(mac.encode().replace(b':', b''))
+
+
+class BroadlinkRM(MediaPlayerDevice):
+    """Representation of a media device."""
+
+    def __init__(self, link, config):
+        """Initialize device."""
+        super().__init__()
+
+        self._support = get_supported_by_config(config)
+        self._config = config
+        self._link = link
+        self._state = STATE_OFF
+        self._source = None
+
+    async def send(self, command):
+        """Send b64 encoded command to device."""
+        if command is None:
+            raise Exception('No command defined!')
+
+        packet = b64decode(command)
+        await self.hass.async_add_job(self._link.send_data, packet)
+
+    @property
+    def name(self):
+        """Return the name of the controlled device."""
+        return self._config.get(CONF_NAME)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return self._support
+
+    async def async_turn_on(self):
+        """Turn on media player."""
+        await self.send(self._config.get(CONF_COMMAND_ON))
+        self._state = STATE_ON
+
```

Comment:
Is state supposed to be come ON even if no CONF_COMMAND_ON has been defined ?

Reply:
Well, it won't have the support turn on feature bit flag set. So theoretically nobody will trigger it. Won't matter to much if it switches state.